conan/tools/meson/toolchain.py

```diff
@@ -319,17 +301,12 @@ def _cross_content(self):
         context['build_machine'] = build_machine
         context['host_machine'] = host_machine
         context['target_machine'] = target_machine
-        return self._render(self._cross_file_template, context)
-
-    def _write_native_file(self):
-        save(self.native_filename, self._native_content)
-
-    def _write_cross_file(self):
-        save(self.cross_filename, self._cross_content)
+        return Template(self._cross_file_template).render(context)
 
     def generate(self):
         if cross_building(self._conanfile):
-            self._write_cross_file()
+            save(self.cross_filename, self._cross_content)
```

Comment:
for me, it makes code harder to read
you always can inline all the methods, e.g. insert the content of the function into the calling function, but it doesn't always make the code easier to read.
I would say, for me, it's a good practice if the function does just one thing which is easily deduced from the function name. it's okay if some functions take one line or less (like tangent is always just a division of sine by cosine, but I'd like to write just tangent because it's a function on its own). same for meson, I prefer to have methods like `write_native_file` and `write_cross_file` which are self-descriptive, instead of reading the internals. at this level, I don't care how the writing of cross-file is exactly implemented (e.g. does it render a template), I prefer to understand the high level logic (write cross file if cross building, write native file otherwise)